Vic2ToHoI4/Source/HOI4World/WarCreator/HoI4WarCreator.cpp

```diff
@@ -72,39 +72,47 @@ void HoI4WarCreator::addAllTargetsToWorldTargetMap()
 }
 
 
-void HoI4WarCreator::addTargetsToWorldTargetMap(shared_ptr<HoI4::Country> country)
+void HoI4WarCreator::addTargetsToWorldTargetMap(std::shared_ptr<HoI4::Country> country)
 {
 	if (country->getGovernmentIdeology() != "democratic")
 	{
 		int maxGCWars = 0;
-		for (auto GC: getDistancesToGreatPowers(country))
+		for (const auto& greatPowerTag: mapUtils.getGPsByDistance(*country, theWorld->getGreatPowers()))
 		{
-			if (maxGCWars < 2)
+			auto greatPower = theWorld->findCountry(greatPowerTag);
+			if (greatPower == nullptr)
 			{
-				string HowToTakeGC = HowToTakeLand(GC.second, country, 3);
-				if (HowToTakeGC == "noactionneeded" || HowToTakeGC == "factionneeded" || HowToTakeGC == "morealliesneeded")
+				continue;
+			}
+
+			if (maxGCWars > 1)
+			{
+				continue;
+			}
+
+			std::string HowToTakeGC = HowToTakeLand(greatPower, country, 3);
+			if (HowToTakeGC == "noactionneeded" || HowToTakeGC == "factionneeded" || HowToTakeGC == "morealliesneeded")
+			{
+				if (greatPower != country)
 				{
-					if (GC.second != country)
+					auto relations = country->getRelations(greatPower->getTag());
+					if ((relations) && (relations->getRelations() < 0))
 					{
-						auto relations = country->getRelations(GC.second->getTag());
-						if ((relations) && (relations->getRelations() < 0))
+						std::vector<std::shared_ptr<HoI4::Country>> tempvector;
+						if (WorldTargetMap.find(greatPower) == WorldTargetMap.end())
 						{
-							vector<shared_ptr<HoI4::Country>> tempvector;
-							if (WorldTargetMap.find(GC.second) == WorldTargetMap.end())
-							{
+							tempvector.push_back(country);
+							WorldTargetMap.insert(std::make_pair(greatPower, tempvector));
+						}
+						if (WorldTargetMap.find(greatPower) != WorldTargetMap.end())
```

Comment:
else, not if? You just inited the WorldTargetMap above because it was dry for that GP, so this is likely an else situation?

Reply:
Sloppy coding, in the WarCreator? It's more likely than you think. And yet again, will be addressed in the future.